Libplanet.Net/Consensus/ConsensusContext.cs

```diff
@@ -120,13 +129,31 @@ public partial class ConsensusContext<T> : IDisposable
         /// of current <see cref="Context{T}"/>, or otherwise returns
         /// <see cref="Libplanet.Net.Consensus.Step.Null"/>.
         /// </returns>
-        public Step Step => _contexts.ContainsKey(Height) ? _contexts[Height].Step : Step.Null;
+        public Step Step
+        {
+            get
+            {
+                lock (_contextLock)
+                {
+                    return _contexts.ContainsKey(Height) ? _contexts[Height].Step : Step.Null;
+                }
+            }
+        }
 
         /// <summary>
         /// A dictionary of <see cref="Context{T}"/> for each heights. Each key represents the
         /// height of value, and value is the <see cref="Context{T}"/>.
         /// </summary>
-        internal Dictionary<long, Context<T>> Contexts => _contexts;
+        internal Dictionary<long, Context<T>> Contexts
+        {
+            get
+            {
+                lock (_contextLock)
+                {
+                    return _contexts;
+                }
+            }
+        }
```

Comment:
This change isn't necessary. `lock`s are there to lock the state of `_contexts`. This is only to return the reference to `_contexts`. 😗